collector/pg_stat_progress_vacuum.go

```diff
@@ -0,0 +1,202 @@
+// Copyright 2025 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"context"
+	"database/sql"
+	"log/slog"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const progressVacuumSubsystem = "stat_progress_vacuum"
+
+func init() {
+	registerCollector(progressVacuumSubsystem, defaultDisabled, NewPGStatProgressVacuumCollector)
+}
+
+type PGStatProgressVacuumCollector struct {
+	log *slog.Logger
+}
+
+func NewPGStatProgressVacuumCollector(config collectorConfig) (Collector, error) {
+	return &PGStatProgressVacuumCollector{log: config.logger}, nil
+}
+
+var (
+	statProgressVacuumPhase = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, progressVacuumSubsystem, "phase"),
+		"Current vacuum phase (numeric). Use label mapping externally for human-readable values.",
+		[]string{"datname", "relname"},
+		nil,
+	)
+
+	statProgressVacuumHeapBlksTotal = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, progressVacuumSubsystem, "heap_blks_total"),
+		"Total number of heap blocks in the table being vacuumed.",
```

Comment:
Looking at the help text, is this a counter or a current status? If it's a current status, we need to remove the `_total` suffix and change it to a GaugeValue.

```suggestion
		prometheus.BuildFQName(namespace, progressVacuumSubsystem, "heap_blks"),
		"Total number of heap blocks in the table being vacuumed.",
```

Reply:
Good catch - this is a current status. It's possible that at the end of a VACUUM, Postgres may in certain conditions be able to truncate empty pages at the end of the table. So I should update this to be a gauge.

Reply:
Updated it to be gauge.